packages/nuxt/src/app/composables/cookie.ts

```diff
@@ -61,7 +61,10 @@ export function useCookie<T = string | null | undefined> (name: string, _opts?:
     const callback = () => {
       if (opts.readonly || isEqual(cookie.value, cookies[name])) { return }
       writeClientCookie(name, cookie.value, opts as CookieSerializeOptions)
-      channel?.postMessage(opts.encode(cookie.value as T))
+
+      const rawCookie = typeof cookie.value === "string" ? cookie.value : JSON.stringify(cookie.value)
+      cookies[name] = destr(rawCookie)
```

Comment:
We have to create new object here to avoid old object reference. Otherwise objects will be equal.

Reply:
Would klona be a better idea, as we already use it above?